lib/sync/meson.build

```diff
@@ -0,0 +1,38 @@
+libephysync_sources = [
+  'ephy-sync-crypto.c',
+  'ephy-sync-service.c',
+  'ephy-synchronizable-manager.c',
+  'ephy-synchronizable.c'
+]
+
+libephysync_deps = [
+  glib_dep,
+  gmp_dep,
+  hogweed_dep,
+  json_glib_dep,
+  libsecret_dep,
+  libsoup_dep,
+  nettle_dep,
+  webkit2gtk_dep
+]
+
+libephysync_includes = include_directories(
+  '.',
+  '..',
+  '../history',
+  '../widgets',
+  '../..',
+  '../../embed'
```

Comment:
I would leave EphyOpenTabsManager where it is, and remove the dependency on EphyEmbedContainer and EphyEmbed. The easiest thing would be to add a new GObject interface under lib/sync, say EphyTabList. Perhaps you can come up with a better name than that. Have EphyWindow implement it. You just need one function, really, ephy_tab_list_get_open_tabs. It would return a GList of structs containing a page title and a URL. That's all you need, I think. Hardest part is coming up with good names and making sure it's the best design.